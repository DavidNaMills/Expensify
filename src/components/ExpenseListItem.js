import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';


const ExpenseListItem = ({description, amount, createdAt, id, dispatch})=>(
    <div>
        <Link to={`/edit/${id}`} ><h4>{description}</h4></Link>
        <p>Amount: {amount}</p>
        <p>Created at: {createdAt}</p>
    </div>
);

//const mapStoreToProps = (state)=>{};



export default ExpenseListItem;