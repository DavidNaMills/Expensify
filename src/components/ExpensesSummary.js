import React from 'react';
import {connect} from 'react-redux';
import expensesTotal from '../selectors/expenses-total';
import getVisibleExpenses from '../selectors/expenses';
import numeral from 'numeral';


export const ExpenseSummary =({expenseCount, expenseTotal})=>{
    return (
            <div>
                <p>viewing {expenseCount}</p>
                <p>grand total is: {numeral(expenseTotal/100).format('$0,0.00')}</p>
            </div>
    );
};


const mapStateToProps =(state)=>{
    const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);

    return {
        expenseCount:visibleExpenses.length,
        expenseTotal: expensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpenseSummary);