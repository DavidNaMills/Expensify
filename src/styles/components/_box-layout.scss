.box-layout{
    background: url('/images/bg.jpg');
    background-size: cover;
    display:flex;
    justify-content: center;
    align-items: center;
    height: 98vh;
    width: 99vw;
}

.box-layout__box{
    background: fade-out($color: white, $amount: 0.25);
    border-radius:3px;
    text-align: center;
    width: 25rem;
    padding: $large-size $medium-size;
}

.box-layout__title{
    margin: 00 $medium-size 0;
    line-height: 1;
    
}