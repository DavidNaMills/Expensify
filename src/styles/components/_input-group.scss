.input-group{
    display:flex;
    flex-direction: column;
    margin-bottom: $medium-size;
    
    @media(min-width: $responsiveSize){
        flex-direction: row;
        margin-bottom: $large-size;
    }
}

.input-group__item{
    margin-bottom: $s-size;
    
    @media(min-width: $responsiveSize){
        margin: 0 $s-size 0 0;

    }
}