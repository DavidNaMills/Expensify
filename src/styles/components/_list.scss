.list-header{
    background-color: $off-white;
    border: 1px solid darken($off-white, 7%);
    color: $gray;
    display:flex;
    justify-content: space-between;
    padding: $s-size $medium-size;
}

.list-body{
    margin-bottom: $medium-size;
    @media(min-width: $responsiveSize){
        margin-bottom: $large-size;
    }
}

.list-item{
    border: 1px solid darken($off-white, 7%);
    border-top: none;
    color: $dark-gray;
    display: flex;
    flex-direction: column;
    padding: $s-size;
    text-decoration: none;
    transition: background .3s ease;
    &:hover{
        background: $off-white;
    }

    @media(min-width: $responsiveSize){
        align-items: center;
        flex-direction: row;
        justify-content: space-between;
        padding: $medium-size;
    }
}

.list-item__message{
    align-items: center;
    color: $gray;
    justify-content: center;
    padding: $medium-size;
    &:hover{
        background: none;
    }
}

.list-item__title{
    margin:0;
    word-break: all;
}

.list-item__sub-title{
    color: $gray;
    font-size: $font-size-small;
}

.list-item__data{
    margin: $s-size 0 0 0;
    @media(min-width: $responsiveSize){
        margin:0;
        padding-left: $s-size;
    }
}