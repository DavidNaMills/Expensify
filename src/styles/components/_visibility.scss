.show-for-mobile{
    @media(min-width: $responsiveSize){
        display:none;
    }
}

.show-for-desktop{
    @media(max-width: $responsiveSize - 0.1rem){
        display:none;
    }    
}