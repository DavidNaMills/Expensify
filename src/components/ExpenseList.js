import {connect} from 'react-redux';
import React from 'react';

import ExpenseListItem from './ExpenseListItem';
import getVisibleExpenses from '../selectors/expenses';

export const ExpenseList = (props)=>(
    <div>
        {
            props.expenses.length === 0? (
                <h1>No Items</h1>
            ):(
               props.expenses.map((item)=>{
                    return <ExpenseListItem key={item.id}{...item} />
            }))
        }
    </div>
);

const mapStateToProps =(state)=>{
    return {
        expenses:getVisibleExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseList);