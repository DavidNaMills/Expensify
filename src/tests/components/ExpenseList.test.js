import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseList} from '../../components/ExpenseList';
import expenses from '../fixtures/expenses';

test('should render expense list with expenses', ()=>{
    const wrapper = shallow(<ExpenseList expenses={expenses}/>);
    expect(wrapper).toMatchSnapshot();
});


test('should render an expense list without items', ()=>{
    const wrapper = shallow(<ExpenseList expenses={[]}/>);
    expect(wrapper).toMatchSnapshot();   
});

