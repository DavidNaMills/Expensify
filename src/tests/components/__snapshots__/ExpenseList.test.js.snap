// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an expense list without items 1`] = `
<div>
  <h1>
    No Items
  </h1>
</div>
`;

exports[`should render expense list with expenses 1`] = `
<div>
  <ExpenseListItem
    amount={999}
    createdAt={0}
    description="orange"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={555}
    createdAt={-345600000}
    description="banana"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={1111}
    createdAt={345600000}
    description="banana"
    id="3"
    key="3"
    note=""
  />
</div>
`;
